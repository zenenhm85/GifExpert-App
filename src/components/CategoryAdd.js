import React, { useState } from "react";
import PropTypes from "prop-types";

export const CategoryAdd = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((categories) => {
        const find = categories.find((item) => item === inputValue);        
        if (!find) {
          setInputValue("");
          return [inputValue, ...categories];
          // setCategories([...categories,`Hunter ${categories.length}`]); This way is also good
        } else {            
            return categories;
        }
      });
      
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        className="input-categories"
        value={inputValue}
        onChange={handleInputChange}
        placeholder="Insert one category"
      />
    </form>
  );
};
CategoryAdd.propTypes = {
    setCategories: PropTypes.func.isRequired,
};