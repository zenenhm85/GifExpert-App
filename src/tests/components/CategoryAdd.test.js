import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";

import { CategoryAdd } from "../../components/CategoryAdd";

describe("Tests in CategoryAdd", () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<CategoryAdd setCategories={setCategories} />);
  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<CategoryAdd setCategories={setCategories} />);
  });

  test("should display the component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("The submit simulation, should not submit if not exist inputValue(the preventFDefault is OK)", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test("The submit should work as expected", () => {
    const value = "Hello Word";
    //Comprobamos que se haya llamado a setCategories uan vez y como parametro una funcion
    //ademas que se limpie la caja de texto
    wrapper.find("input").simulate("change", { target: { value } });
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    expect(wrapper.find("input").prop("value")).toBe("");
  });
  test("The submit not work if length of value is less than 3", () => {
    const value = "Hi";
    wrapper.find("input").simulate("change", { target: { value } });
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });
});
