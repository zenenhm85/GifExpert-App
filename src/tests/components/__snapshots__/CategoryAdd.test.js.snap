// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in CategoryAdd should display the component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="input-categories"
    onChange={[Function]}
    placeholder="Insert one category"
    type="text"
    value=""
  />
</form>
`;
